client_delivery/shared/src/commonMain/kotlin/presentation/base/BaseScreenModel.kt

```diff
@@ -71,7 +75,16 @@ abstract class BaseScreenModel<S, E>(initialState: S) : ScreenModel, KoinCompone
         return inScope.launch(dispatcher) {
             try {
                 function()
+            } catch (e: InvalidUsernameException) {
+                onError(ErrorState.InvalidUsername)
+            } catch (e: InvalidPasswordException) {
+                onError(ErrorState.InvalidPassword)
+            } catch (e: UserNotFoundException) {
+                onError(ErrorState.UserNotFound)
+            } catch (e: UnAuthorizedException) {
+                onError(ErrorState.UnAuthorized)
             } catch (e: Exception) {
+                println("mohamed: $e")
```

Comment:
u can clone the way we handled exceptions in this PR #364 
may in another PR but that way is better.